core/src/test/java/org/apache/iceberg/TestSchemaID.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg;
+
+import java.util.List;
+import java.util.stream.IntStream;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.apache.iceberg.types.Types;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import static org.apache.iceberg.TestHelpers.assertSameSchemaMap;
+import static org.apache.iceberg.types.Types.NestedField.optional;
+import static org.apache.iceberg.types.Types.NestedField.required;
+
+@RunWith(Parameterized.class)
+public class TestSchemaID extends TableTestBase {
+
+  @Parameterized.Parameters(name = "formatVersion = {0}")
+  public static Object[] parameters() {
+    return new Object[] { 1, 2 };
+  }
+
+  public TestSchemaID(int formatVersion) {
+    super(formatVersion);
+  }
+
+  @Test
+  public void testNoChange() {
+    // add files to table
+    table.newAppend().appendFile(FILE_A).appendFile(FILE_B).commit();
+
+    validateSingleSchema();
+    validateSnapshotsAndHistoryEntries(ImmutableList.of(0));
+
+    // remove file from table
+    table.newDelete().deleteFile(FILE_A).commit();
+
+    validateSingleSchema();
+    validateSnapshotsAndHistoryEntries(ImmutableList.of(0, 0));
+
+    // add file to table
+    table.newFastAppend().appendFile(FILE_A2).commit();
+
+    validateSingleSchema();
+    validateSnapshotsAndHistoryEntries(ImmutableList.of(0, 0, 0));
+  }
+
+  @Test
+  public void testSchemaIdChangeInSchemaUpdate() {
+    // add files to table
+    table.newAppend().appendFile(FILE_A).appendFile(FILE_B).commit();
+
+    validateSingleSchema();
+    validateSnapshotsAndHistoryEntries(ImmutableList.of(0));
+
+    // cache old schema
+    Schema oldSchema = table.schema();
+
+    // update schema
+    table.updateSchema().addColumn("data2", Types.StringType.get()).commit();
+
+    Schema updatedSchema = new Schema(1,
```

Comment:
Can't we just get the table's current schema? Or is this intended to check that schema evolution produces the right ID in addition to the snapshot tracking?

Reply:
Yeah I was hoping to make the check explicitly that schema id is now different